Sources/ArcGISToolkit/Components/JobManager/JobManager.swift

```diff
@@ -0,0 +1,301 @@
+/// status of the work on the server can be checked intermittently.
+///
+/// To enable polling while an app is backgrounded, the job manager will request from the system a
+/// background refresh task (if enabled via the ``JobManager/preferredBackgroundStatusCheckSchedule``
+/// property). If the system later executes the background refresh task then the
+/// job manager will check the status of any running jobs. At that point the jobs may start
+/// downloading their result. Note, this does not work on the simulator, this behavior can only
+/// be tested on an actual device.
+///
+/// Now that the job can check status in the background, it can start downloading in the background.
+/// By default, jobs will download their results with background URL session. This means that the
+/// download can execute out of process, even if the app is terminated. If the app is terminated and
+/// then later relaunched by the system because a background downloaded completed, then you may
+/// call the ``JobManager/resumeAllPausedJobs()`` method from the application relaunch point,
+/// which will correllate the jobs to their respective downloads that completed and the jobs will
+/// then finish. The app relaunch point can happen via the SwiftUI modifier `.backgroundTask(.urlSession(...))`.
+/// In UIKit it would be the `UIApplicationDelegate` method `func application(UIApplication, handleEventsForBackgroundURLSession: String, completionHandler: () -> Void)`
+@MainActor
+public class JobManager: ObservableObject {
+    /// The shared job manager.
+    public static let `shared` = JobManager()
+    
+    /// The jobs being managed by the job manager.
+    @Published
+    public var jobs: [any JobProtocol] = []
+    
+    /// The key for which state will be serialized under the user defaults.
+    private var defaultsKey: String {
+        return "com.esri.ArcGISToolkit.jobManager.jobs"
+    }
+    
+    /// The preferred schedule for performing status checks while the application is in the
+    /// background. This allows an application to check to see if jobs have completed and optionally
+    /// post a local notification to update the user. The default value is `disabled`.
+    /// When the value of this property is not `disabled`, this setting is just a preference.
+    /// The operating system ultimately decides when to allow a background task to run.
+    /// If you enable background status checks then you must also make sure to have enabled
+    /// "Background Fetch" and "Background Processing" background modes in your application settings.
+    /// You must also add "com.esri.ArcGISToolkit.jobManager.statusCheck" to the "Permitted background task scheduler identifiers"
+    /// in your application's plist file. Note, this also only works on device and not on the simulator.
+    /// More information can be found here: https://developer.apple.com/documentation/backgroundtasks/refreshing_and_maintaining_your_app_using_background_tasks
```

Comment:
Link is un-clickable rendering as text:

<img width="516" alt="Screenshot 2023-08-31 at 10 07 03" src="https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/assets/16397058/a75765d7-4550-4fd8-b853-78c0a2ac0196">


```suggestion
    /// More information can be found [here](https://developer.apple.com/documentation/backgroundtasks/refreshing_and_maintaining_your_app_using_background_tasks).

```